src/deploy/firestore/prepare.ts

```diff
@@ -65,6 +80,30 @@ export default async function (context: any, options: any): Promise<void> {
     context.firestoreRules = true;
   }
 
-  prepareIndexes(context, options);
-  await prepareRules(context, options);
+  const firestoreConfigs: fsConfig.ParsedFirestoreConfig[] = fsConfig.getFirestoreConfig(
+    context.projectId,
+    options
+  );
+  if (!firestoreConfigs || firestoreConfigs.length === 0) {
+    return;
+  }
+
+  context.firestore = context.firestore || {};
+  context.firestore.indexes = [];
+  context.firestore.rules = [];
+  const rulesDeploy: RulesDeploy = new RulesDeploy(options, RulesetServiceType.CLOUD_FIRESTORE);
+  _.set(context, "firestore.rulesDeploy", rulesDeploy);
```

Comment:
This _.set is not necessary because 3 lines above you ensure that context.firestore exists.
```suggestion
  context.firestore.rulesDeploy = rulesDeploy;
```

Reply:
This is also the last use of lodash in the file i beleive, so you can remove it's import too 🧹